src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -172,7 +167,13 @@ internal void Abort()
                         // interpreted correctly by the server if it's in the middle of a
                         // DATA command or some similar situation.  This may send a RST
                         // but this is ok in this situation.  Do not reuse this connection
-                        _tcpClient.LingerState = new LingerOption(true, 0);
+                        try
+                        {
+                            _tcpClient.LingerState = new LingerOption(true, 0);
+                        }
+                        catch
```

Comment:
@stephentoub should I change the PR so we can figure out what exact exception gets thrown here? Or do we keep the general catch?

Reply:
My preference would be to catch the most specific exception possible, though the risk here is admittedly minimal.

Reply:
Is the exception something other than derived from System.Exception? Otherwise, why not just write `catch (Exception)` ?

But I agree it is useful to understand what's going on here. I would expect the exception is probably a SocketException.

Reply:
> should I change the PR so we can figure out what exact exception gets thrown here?

My preference is to catch just the exception types we expect.  Thanks.

Reply:
@scalablecory @davidsh @scalablecory I have to change the PR so we know what exception gets thrown here.

I'll revert this try-catch, and also remove the general try-catch in `SmtpClient.Abort`. That should surface the exception.

We can give it a couple of test runs here. But since the issue doesn't happen often, probably we'll have to merge and wait for some time until it shows up in CI.